tip/environments.py

```diff
@@ -0,0 +1,85 @@
+import os
+import json
+
+from tip.util import parse_package_specifier
+
+
+def locate(tip_dir: str, name: str) -> str:
+    """Find file of the environment called `name`."""
+    return os.path.join(_get_environments_dir(tip_dir), f"{name}.json")
+
+
+def read_environment_by_path(path: str) -> dict:
+    """Get package list of the environment at `path`."""
+    if not os.path.isfile(path):
+        raise FileNotFoundError(f"Environment file is not found at {path}")
+    with open(path, mode='r', encoding='utf8') as environment_file:
+        return json.load(environment_file)
+
+
+def read_environment_by_name(tip_dir: str, name: str) -> dict:
+    """Get package list of the environment called `name`."""
+    path = locate(tip_dir, name)
+    return read_environment_by_path(path)
+
+
+def save_environment(environment: dict | None, path: str, /, *, rewrite: bool = False):
+    """Save environment to path, rewriting the file if `rewrite` is True."""
+    if os.path.isfile(path) and not rewrite:
+        raise RuntimeError("Environment file exists and asked not to rewrite")
+    with open(path, mode='w+', encoding='utf8') as environment_file:
+        json.dump({} if environment is None else environment, environment_file)
+
+
+def exists(tip_dir: str, name: str) -> bool:
+    """Check if file of the environment called `name` exists."""
+    path = locate(tip_dir, name)
+    return os.path.isfile(path)
+
+
+def add_to_environment_at_path(package: str, path: str, replace: bool = False):
```

Comment:
This function as well as some other functions in this PR have too much of responsibility. It performs the following operations:
1. Load environment.
2. Add package.
3. Save environment.

What if I want to add package to preloaded environment? Why not to implement Environment class such as 👇?
```python
class Environment:
    @staticmethod
    def load(path=None, *, name=None): ...
    
    @staticmethod
    def locate(name): ...
    
    def save(self): ...
    
    def add_package(self, package_specifier): ...
    
    ...
```